libs/blocks/modal/modal.js

```diff
@@ -65,9 +68,21 @@ async function getPathModal(path, dialog) {
   const block = createTag('a', { href: path });
   dialog.append(block);
 
+  // eslint-disable-next-line import/no-cycle
   const { default: getFragment } = await import('../fragment/fragment.js');
   await getFragment(block);
 }
+function sendViewportDimensionsToiFrame(source) {
+  const viewportWidth = Math.max(document.documentElement.clientWidth || 0, window.innerWidth || 0);
+  source.postMessage({ mobileMax: MOBILE_MAX, tabletMax: TABLET_MAX, viewportWidth }, '*');
+}
+
+export function sendViewportDimensionsOnRequest(messageInfo) {
+  if (messageInfo.data === 'viewportWidth') {
+    sendViewportDimensionsToiFrame(messageInfo.source);
+    window.addEventListener('resize', () => sendViewportDimensionsToiFrame(messageInfo.source));
+  }
+}
```

Comment:
This would be the quickest solution - though if you can move the `messageInfo.data === 'viewportWidth'` condition to the eventListener on L157, that's cleaner and you don't need the extra boolean check if the event listener already exists.
```suggestion
let resizeListenerAdded = false;
export function sendViewportDimensionsOnRequest(messageInfo) {
  if (messageInfo.data === 'viewportWidth') {
    sendViewportDimensionsToiFrame(messageInfo.source);
    if (!resizeListenerAdded) {
      window.addEventListener('resize', debounce(() => sendViewportDimensionsToiFrame(messageInfo.source), 50));
      resizeListenerAdded = true;
    }
  }
}
```